example/cnn_mnist_1d.f90

```diff
@@ -0,0 +1,67 @@
+program cnn_mnist_1d
+
+    use nf, only: network, sgd, &
+      input, conv1d, maxpool1d, flatten, dense, reshape, reshape2d, locally_connected_1d, &
+      load_mnist, label_digits, softmax, relu
+  
+    implicit none
+  
+    type(network) :: net
+  
+    real, allocatable :: training_images(:,:), training_labels(:)
+    real, allocatable :: validation_images(:,:), validation_labels(:)
+    real, allocatable :: testing_images(:,:), testing_labels(:)
+    integer :: n
+    integer, parameter :: num_epochs = 250
+  
+    call load_mnist(training_images, training_labels, &
+                    validation_images, validation_labels, &
+                    testing_images, testing_labels)
+  
+    net = network([ &
+      input(784), &
+      reshape2d([28,28]), &
+      locally_connected_1d(filters=8, kernel_size=3, activation=relu()), &
```

Comment:
Why is a `reshape2d` needed if `locally_connected_1d` is used?

Reply:
Ok. I guess that the `reshape2d` has a similar function to the `input_shape` in `keras`